cnf-certification-test/operator/suite.go

```diff
@@ -104,23 +108,28 @@ func testOperatorInstallationWithoutPrivileges(env *provider.TestEnvironment) {
 		}
 
 		if badRuleFound {
-			badOperators = append(badOperators, env.Operators[i].String())
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewOperatorReportObject(env.Operators[i].Namespace, env.Operators[i].Name, "Operator has privileges on cluster resources ", false))
+		} else {
+			compliantObjects = append(compliantObjects, testhelper.NewOperatorReportObject(env.Operators[i].Namespace, env.Operators[i].Name, "Operator has no privileges on cluster resources", true))
 		}
 	}
 
-	testhelper.AddTestResultLog("Non-compliant", badOperators, tnf.ClaimFilePrintf, ginkgo.Fail)
+	testhelper.AddTestResultReason(compliantObjects, nonCompliantObjects, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
 func testOperatorOlmSubscription(env *provider.TestEnvironment) {
-	nonCompliantCsvs := []string{}
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
 
 	for i := range env.Operators {
 		operator := env.Operators[i]
 		if operator.SubscriptionName == "" {
 			tnf.ClaimFilePrintf("OLM subscription not found for operator from csv %s", provider.CsvToString(operator.Csv))
-			nonCompliantCsvs = append(nonCompliantCsvs, provider.CsvToString(operator.Csv))
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewOperatorReportObject(env.Operators[i].Namespace, env.Operators[i].Name, "OLM subscription not found for operator, so it is not installed via OLM", false))
```

Comment:
Add a subscription name field to the operator reporting object and use the addField function to add it